src/components/shared/WithSize.js

```diff
@@ -36,23 +36,23 @@ export function withSize<
   $ReadOnly<$Diff<Props, SizeProps>>
 > {
   return class WithSizeWrapper extends React.PureComponent<*, State> {
-    _resizeListener: Event => void;
+    _resizeListener: () => void;
     state = { width: 0, height: 0 };
 
-    _observeSize = (wrappedComponent: React.Component<any> | null) => {
-      if (!wrappedComponent) {
-        return;
-      }
-      const container = findDOMNode(wrappedComponent); // eslint-disable-line react/no-find-dom-node
+    componentDidMount() {
+      const container = findDOMNode(this); // eslint-disable-line react/no-find-dom-node
       if (!container) {
         throw new Error('Unable to find the DOMNode');
       }
       this._resizeListener = () => {
         this._updateWidth(container);
       };
       window.addEventListener('resize', this._resizeListener);
-      this._updateWidth(container);
-    };
+
+      // Wrapping the first update in a requestAnimationFrame to defer the
+      // calculation until the full render is done.
+      requestAnimationFrame(() => this._updateWidth(container));
```

Comment:
This is the only needed change, but I thought it was a good opportunity to get rid of the ref. (we still need this `findDOMNode` though)

Reply:
Without this change, the call to `getBoundingClientRect` is done too early, likely triggering a sync reflow too, I think even before the sidebar is added to the DOM. I believe this is because the ref callback is called synchronously as soon as the element is added to the DOM. `componentDidMount` has the same issue but is a cleaner way as it's a more appropriate lifecycle method.
Using `requestAnimationFrame` waits that everything is rendered before getting the measurement. Maybe `setTimeout` would be better?